src/components/Overview/PlotlyCharts/ForceDirectedGraph.tsx

```diff
@@ -1,18 +1,181 @@
+  AvastarParsedDataPoint,
+  AvastarParsedDataPointState,
+} from '../../../types/dataTypes';
+
 const { useRef } = React;
 
 export default function ForceGraph() {
+  // Fetch data from State
+  const avastarParsedData: readonly AvastarParsedDataPoint[] = useSelector(
+    (state: AvastarParsedDataPointState) => state.avastarParsedData,
+    shallowEqual
+  );
+
+  // Filter avastarParsedData by selecting only the objects that are relevant for the force graph
+
+  // Define list of action_type to filter the data object stored in state
+  const actions = [
+    'Advertisers whose ads you have recently seen or clicked on',
+    'Advertiser who has added your name to an audience based on your information or your activity outside of Facebook',
+    'Company that shared information with facebook about your activity on their website/application',
+    'Apps that have been installed on your mobile device',
+    'Account linked to your Facebook profile',
+    'Game you played on Facebook',
+  ];
+
+  // Filter data object based on the list defined above
+  let filteredData = [];
+  for (let i = 0; i < avastarParsedData.length; i++) {
+    if (actions.includes(avastarParsedData[i].action_type)) {
+      filteredData.push(avastarParsedData[i]);
+    }
+  }
+
+  // Pick randomly 150 data points coming from the filtered data object
+  // @ts-ignore
+  var data = filteredData.sort(() => 0.5 - Math.random()).slice(0, 150);
+
+  // Transform filtered data object into the right shape for the force graph (nodes and links)
+  const setForceGraphData = () => {
+    // Set empty arrays
+    var forceGraphData = [] as any;
+    var nodes = [] as any;
+    var links = [] as any;
+    // Set the initial nodes (You, Facebook, Google)
+    nodes.push({
+      id: 'You',
+      group: 1,
+      color: '#ff6961',
+    });
+    nodes.push({
+      id: 'facebook',
+      group: 2,
+      color: '#59adf6',
+    });
+    nodes.push({
+      id: 'google',
+      group: 3,
+      color: '#42d6a4',
+    });
+    // Set the initial links (You - Facebook, You - Google)
+    links.push({
+      source: 'You',
+      target: 'facebook',
+      value: 10, // intensity of the moving particles defined relatively by the volume of data transferred
+    });
+    links.push({
+      source: 'You',
+      target: 'google',
+      value: 10,
+    });
+
+    // Define function to transform every object that have been filtered
+    const transformData = (
+      initialObject: AvastarParsedDataPoint[],
+      nodesObject: object[],
+      linksObject: object[],
+      forceGraphInput: object[],
+      actionType: string,
+      groupId: number,
+      colorCode: string
+    ) => {
+      for (let i = 0; i < initialObject.length; i++) {
+        if (initialObject[i].action_type === actionType) {
+          nodesObject.push({
+            id: data[i].details,
+            group: groupId,
+            color: colorCode,
+          });
+          linksObject.push({
+            source: 'You',
```

Comment:
Est-ce utile de relier systématiquement les third parties à soi ? Par ex, les ads sur lesquels on clique depuis Facebook + jeux installés depuis Facebook, est-ce qu'on pourrait pas uniquement les lier via Facebook, histoire de mieux représenter les choix de l'utilisateur quant à ses interactions directes ?

Reply:
+1 pour la lisibilité et la compréhension je trouve ça mieux perso. Tu as une raison @tangigouez de pourquoi tu n'as pas fait comme ça ?

Reply:
Oui je suis d'accord.

Voilà ce que je propose : 

Entreprises avec lesquelles l'utilisateur entretient une relation "directe": 
- 'Company that shared information with facebook about your activity on their website/application',
- 'Apps that have been installed on your mobile device',
- 'Account linked to your Facebook profile',

Entreprises avec lesquelles l'utilisateur entretient une relation "INdirecte": 
- 'Advertisers whose ads you have recently seen or clicked on',
- 'Advertiser who has added your name to an audience based on your information or your activity outside of Facebook',
- 'Game you played on Facebook',

Ce que ça donne 👇 

![Screenshot 2022-05-31 at 22 26 12](https://user-images.githubusercontent.com/45219966/171279077-29abb1e5-f849-4820-acc7-d810111b18d0.png)